app/src/main/java/com/github/se/gomeet/MainActivity.kt

```diff
@@ -86,6 +90,36 @@ class MainActivity : ComponentActivity() {
             composable(Route.PUBLIC_CREATE) {
               CreateEvent(navAction, EventViewModel(userIdState.value), false)
             }
+
+            composable(
+                route = Route.EVENT_INFO,
+                arguments =
+                    listOf(
+                        navArgument("title") { type = NavType.StringType },
+                        navArgument("date") { type = NavType.StringType },
+                        navArgument("time") { type = NavType.StringType },
+                        navArgument("organizer") { type = NavType.StringType },
+                        navArgument("rating") { type = NavType.FloatType },
+                        navArgument("description") { type = NavType.StringType },
+                        navArgument("latitude") {
+                          type = NavType.FloatType
+                        }, // Change to DoubleType
+                        navArgument("longitude") {
+                          type = NavType.FloatType
+                        } // Change to DoubleType
+                        )) { entry ->
+                  val title = entry.arguments?.getString("title") ?: ""
+                  val date = entry.arguments?.getString("date") ?: ""
+                  val time = entry.arguments?.getString("time") ?: ""
+                  val organizer = entry.arguments?.getString("organizer") ?: ""
+                  val rating = entry.arguments?.getFloat("rating") ?: 0f
+                  val description = entry.arguments?.getString("description") ?: ""
+                  val latitude = entry.arguments?.getFloat("latitude") ?: 0.0
+                  val longitude = entry.arguments?.getFloat("longitude") ?: 0.0
+                  val loc = LatLng(latitude.toDouble(), longitude.toDouble())
+
+                  EventInfoScreen(nav)
+                }
```

Comment:
Have you considered passing an Event object directly as an argument for the navigation? If we introduce an Event parameter to the EventInfoScreen composable, it could streamline the implementation and reduce the amount of code. This approach might enhance maintainability and improve readability by minimizing the need for additional data handling in multiple places.

Reply:
We still need to see how the event attributes are stored. Eg. we store the image as an int (string for the URL) but we need to get it from firebase (Vlad is working on that). Will be looked into in the future in my opinion but I think we could keep it like this for now/